addons/account/migrations/13.0.1.1/post-migration.py

```diff
@@ -604,35 +605,45 @@ def migration_voucher_moves(env):
     )
 
 
-def _move_model_in_data(env, ids_map, old_model, new_model):
+def _move_model_in_data(env, old_model, new_model, field):
     renames = [
         ('mail_message', 'model', 'res_id'),
-        ('mail_followers', 'res_model', 'res_id'),
         ('ir_attachment', 'res_model', 'res_id'),
         ('mail_activity', 'res_model', 'res_id'),
         ('ir_model_data', 'model', 'res_id'),
     ]
-    for old_id, new_id in ids_map:
-        for rename in renames:
-            query = """
-                UPDATE {table}
-                SET {field1} = %(new_value1)s, {field2} = %(new_value2)s
-                WHERE {field1} = %(old_value1)s AND {field2} = %(old_value2)s"""
-            if rename[0] == 'mail_followers':
-                query += """ AND partner_id NOT IN (
-                    SELECT partner_id FROM mail_followers
-                    WHERE res_model = 'account.move' AND res_id = %(new_value2)s
-                )"""
-            env.cr.execute(sql.SQL(query).format(
-                table=sql.Identifier(rename[0]),
-                field1=sql.Identifier(rename[1]),
-                field2=sql.Identifier(rename[2])
-            ), {
-                "old_value1": old_model,
-                "new_value1": new_model,
-                "old_value2": old_id,
-                "new_value2": new_id,
-            })
+    for rename in renames:
+        query = """
+            UPDATE {table} t
+            SET {field1} = %(new_value1)s, {field2} = am.id
+            FROM account_move am
+            WHERE t.{field1} = %(old_value1)s AND am.{field} = t.{field2}"""
+        openupgrade.logged_query(env.cr, sql.SQL(query).format(
+            table=sql.Identifier(rename[0]),
+            field1=sql.Identifier(rename[1]),
+            field2=sql.Identifier(rename[2]),
+            field=sql.Identifier(field)
+        ), {
+            "old_value1": old_model,
+            "new_value1": new_model,
+        })
+    openupgrade.logged_query(env.cr, sql.SQL("""
+        UPDATE mail_followers mf
+            SET res_model = %(new_value1)s, res_id = am.id
+            FROM account_move am
+                JOIN mail_followers mf1
+                    ON am.{field} = mf1.res_id AND mf1.res_model = %(old_value1)s
+                LEFT JOIN mail_followers mf2
+                    ON am.id = mf2.res_id
+                        AND mf2.res_model = 'account.move'
+                        AND mf2.partner_id = mf1.partner_id
```

Comment:
```suggestion
                    ON (am.{field} = mf1.res_id AND mf1.res_model = %(old_value1)s)
                LEFT JOIN mail_followers mf2
                    ON (am.id = mf2.res_id
                        AND mf2.res_model = 'account.move'
                        AND mf2.partner_id = mf1.partner_id)
```
parenthesis not needed, but helps readability :)